### PR TITLE
Bsky appview admin and moderation endpoints

### PR DESCRIPTION
**Pull request by devinivy**
Core admin/moderation tooling is now present in the appview:

 - Implement com.atproto.admin.* methods on appview.
 - Implement com.atproto.moderation.createReport method on appview.
 - Implement com.atproto.identity.resolveHandle method on appview.
 - Update dev-env so that it can be run with a bsky appview.
 - Update appview blob resolver so that it doesn't resolve taken-down blobs.
 - Invalidate image cache on takedown.

Does not have automated labeling or serve labels on non-admin views yet— that will be in a followup PR.

### CODE REVIEW

File: packages/bsky/src/db/tables/label.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const tableName = 'label'
+
+export interface Label {
+  src: string
+  uri: string
+  cid: string
+  val: string
+  neg: 0 | 1 // @TODO convert to boolean in app-view
```
Comment (dholms) [inline packages/bsky/src/db/tables/label.ts:8]: oh this is a boolean now! got it right in migration 👌

File: packages/bsky/src/services/moderation/index.ts [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+  async takedownRepo(info: { takedownId: number; did: string }) {
+    await this.db.db
+      .updateTable('actor')
+      .set({ takedownId: info.takedownId })
+      .where('did', '=', info.did)
+      .where('takedownId', 'is', null)
+      .executeTakeFirst()
+  }
+
+  async reverseTakedownRepo(info: { did: string }) {
```
Comment (dholms) [inline packages/bsky/src/services/moderation/index.ts:291]: should we be invalidating all related blobs here?

File: packages/bsky/src/api/blob-resolver.ts [hunk 3]
```diff
@@ -30,7 +31,24 @@ export const createRouter = (ctx: AppContext): express.Router => {
         return next(createError(400, 'Invalid cid'))
       }
 
-      const { pds } = await ctx.didResolver.resolveAtpData(did) // @TODO cache did info
+      const [{ pds }, takedown] = await Promise.all([
+        ctx.didResolver.resolveAtpData(did), // @TODO cache did info
+        ctx.db.db
+          .selectFrom('moderation_action_subject_blob')
```
Comment (dholms) [inline packages/bsky/src/api/blob-resolver.ts:37]: do you need a `.select()` or `selectAll()` here?

File: packages/bsky/src/api/com/atproto/identity/resolveHandle.ts [hunk 4]
```diff
@@ -13,6 +13,16 @@ export default function (server: Server, ctx: AppContext) {
     if (user) {
       did = user.did
     } else {
+      const publicHostname = ctx.cfg.publicUrl
```
Comment (dholms) [inline packages/bsky/src/api/com/atproto/identity/resolveHandle.ts:16]: we just gotta ensure we don't have any alternate subdomains for the appview 😛

**Review by dholms — APPROVED** [review]
left some comments but overall lookin rad 😎

File: packages/bsky/src/api/com/atproto/identity/resolveHandle.ts (hunk 4 above)
Comment (devinivy) [reply packages/bsky/src/api/com/atproto/identity/resolveHandle.ts:16]: Yeah 😳 I wonder if we should eventually wrap this into a more general configuration allowing multiple hostnames or even wildcards.  I'll leave a note here for now.

File: packages/bsky/src/api/blob-resolver.ts (hunk 3 above)
Comment (devinivy) [reply packages/bsky/src/api/blob-resolver.ts:37]: I double-checked and it does work without it, but good call!  Definitely should have one.

File: packages/bsky/src/services/moderation/index.ts (hunk 2 above)
Comment (devinivy) [reply packages/bsky/src/services/moderation/index.ts:291]: That's a good question.  To date blobs haven't been a part of repo takedowns, but it's possible to start allowing it with a little effort.  Here's where that currently gets stopped: https://github.com/bluesky-social/atproto/blob/f30887bc1c21936ff2b79832d4f80e04edfdb9e2/packages/pds/src/services/moderation/index.ts#L215-L217

**Pull request merged**